scripts/extensions/auto-tagging-widget/src/auto-tagging.tsx

```diff
@@ -395,26 +395,91 @@ export function getAutoTaggingComponent(superdesk: ISuperdesk, label: string) {
 
                                 const entities = items.filter((tag) => isEntity(tag));
                                 const entitiesGrouped = entities.groupBy((tag) => tag?.group.value);
-                                const entitiesGroupedAndSorted = entitiesGrouped.sortBy(
-                                    (_, key) => key!.toString().toLocaleLowerCase(),
-                                    (a, b) => a.localeCompare(b),
-                                );
+
+                                const entitiesGroupedAndSortedByConfig = entitiesGrouped
+                                    .filter((_, key) => hasConfig(key, this.iMatricsFields.entities))
+                                    .sortBy((_, key) => this.iMatricsFields.entities[key].order,
+                                        (a, b) => a - b);
+
+                                const entitiesGroupedAndSortedNotInConfig = entitiesGrouped
+                                    .filter((_, key) => !hasConfig(key, this.iMatricsFields.entities))
+                                    .sortBy((_, key) => key!.toString().toLocaleLowerCase(),
+                                        (a, b) => a.localeCompare(b));
+
+                                const entitiesGroupedAndSorted = entitiesGroupedAndSortedByConfig
+                                    .concat(entitiesGroupedAndSortedNotInConfig);
 
                                 const others = items.filter((tag) => isEntity(tag) === false);
                                 const othersGrouped = others.groupBy((tag) => tag.group.value);
 
-                                const othersGroupedAndSortedByConfig = othersGrouped
-                                    .filter((_, key) => hasConfig(key, this.iMatricsFields))
-                                    .sortBy((_, key) => this.iMatricsFields[key].order,
+                                let allGrouped = OrderedMap<string, any>();
```

Comment:
Avoid using `any`. In this case, the type would be `JSX.Element`. It better thought to avoid putting jsx elements to arrays and render directly. It's easier to keep track of element order this way.